### PR TITLE
Feature/#8 add ci task to check if the generated api is outdated

### LINKED ISSUE #8: Add CI task to check if the generated API is outdated
- The published json description for the API can change without our knowledge
- For that reason, the generated API can get outdated and we need to check if it is up-to-date 
- In the CI we need to run the generation and compare with git status or git diff if we have to update
- This job has to run daily and maybe send a slack message

### PR DESCRIPTION
**Pull request by ckunki**
Closes #8

### CODE REVIEW

File: doc/developer_guide/developer_guide.md [hunk 1]
```diff
@@ -31,8 +26,7 @@ poetry run nox generate-api
 
 ### Change the Source URL of API Model JSON Definition
```
Comment (tkilias) [inline doc/developer_guide/developer_guide.md:27]:
```suggestion
### Change the Source URL of the API Model JSON Definition
```

File: pyproject.toml [hunk 2]
```diff
@@ -83,12 +83,14 @@ ignore-paths = [
 [[tool.mypy.overrides]]
 module = [
     "exasol.toolbox.nox.tasks",
+    "exasol.saas.client.openapi.*",
```
Comment (tkilias) [inline pyproject.toml:86]: Does it complain?

File: noxfile.py [hunk 3]
```diff
@@ -21,5 +21,15 @@ def generate_api(session: Session):
         "update",
         "--url", f"{SAAS_HOST}/openapi.json",
         "--config", "openapi_config.yml",
+        silent=True,
```
Comment (tkilias) [inline noxfile.py:24]: why did you add it?

File: noxfile.py [hunk 4]
```diff
@@ -21,5 +21,15 @@ def generate_api(session: Session):
...
     session.run("isort", "-q", "exasol/saas/client/openapi")
+
+
+@nox.session(name="check-api-outdated", python=False)
+def check_api_outdated(session: Session):
+    """
+    Generate API and run git diff to verify if API is out-dated.
+    """
+    generate_api(session)
+    session.run("git", "diff", "--exit-code")
```
Comment (tkilias) [inline noxfile.py:35]: nice --exit-code I didn't know

**Review by tkilias — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 2 above)
Comment (ckunki) [reply pyproject.toml:86]: Oh, yes!

File: noxfile.py (hunk 3 above)
Comment (ckunki) [reply noxfile.py:24]: I added `silent=True` to keep command line and ci build clean. 
Otherwise you get 177 lines with warnings like 
> WARNING parsing GET /api/v1/accounts/{accountID}/databases within databases.
> Invalid response status code default (not a valid HTTP status code), response will be ommitted from generated client

If you want, we can make the silent optional and set it to `False` only in the CI build.

Comment (tkilias) [reply noxfile.py:24]: https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables

The env variable I mean is `CI`

Comment (ckunki) [reply noxfile.py]: Thanks - fixed.

File: .github/workflows/check-api-outdated.yml [hunk 5]
```diff
@@ -1,6 +1,14 @@
 name: Check if OpenAPI is Outdated
 
 on:
+  push:
+    branches-ignore:
+      - "github-pages/*"
+      - "gh-pages/*"
+      - "main"
```
Comment (tkilias) [inline .github/workflows/check-api-outdated.yml:8]: why shouldn't it run on main

File: .github/workflows/check-api-outdated.yml [hunk 6]
```diff
@@ -1,6 +1,14 @@
...
+      - "master"
+  pull_request:
+    types: [opened, reopened]
```
Comment (tkilias) [inline .github/workflows/check-api-outdated.yml:11]: why do you limit to opened and reopended?

File: noxfile.py [hunk 7]
```diff
@@ -9,17 +10,34 @@
...
     Call openapi-python-client to generate the client api based on the
     openapi specification for Exasol SaaS in JSON format, see
     https://github.com/openapi-generators/openapi-python-client.
+
+    By default run generator silently, Except for CI build, which is detected
+    by environment variable ``CI``, see
+    https://docs.github.com/en/actions/learn-github-actions/variables.
+    #default-environment-variables.
     """
+    silent = False if os.environ.get("CI") else True
```
Comment (tkilias) [inline noxfile.py:26]:
```suggestion
    silent = "CI" in os.environ
```

this will actually fail, when it is not set

**Review by tkilias — CHANGES_REQUESTED** [review]

File: .github/workflows/check-api-outdated.yml (hunk 5 above)
Comment (ckunki) [reply .github/workflows/check-api-outdated.yml:8]: I copied the code from `ci.yml` which was generated by PTB.

File: .github/workflows/check-api-outdated.yml (hunk 6 above)
Comment (ckunki) [reply .github/workflows/check-api-outdated.yml:11]: I copied the code from `ci.yml` which was generated by PTB.

File: noxfile.py (hunk 7 above)
Comment (ckunki) [reply noxfile.py:26]: Actually, the `.get()` did not fail, when the variable was unset, see https://python-reference.readthedocs.io/en/latest/docs/dict/get.html. 

But I prefer your proposal as it is shorter.

File: .github/workflows/check-api-outdated.yml (hunk 5 above)
Comment (ckunki) [reply .github/workflows/check-api-outdated.yml:8]: I think the intention is to run checks only once (on the branch) but not another time (after merging).

File: .github/workflows/check-api-outdated.yml (hunk 6 above)
Comment (ckunki) [reply .github/workflows/check-api-outdated.yml:11]: [GitHub documentation](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) shows 21 activity types for PRs. To me, the 2 named above make sense. 
If at all, then I would look for an activity `updated`. But such is not listed.

Additionally, I experienced the workflow `ci.yml` with identical settings to have been triggered for each push, just as desired. 

In summary, the selected activity types seem OK to me.

Comment (ckunki) [reply .github/workflows/check-api-outdated.yml:10]: See https://github.com/exasol/python-toolbox/issues/160

**Review by tkilias — APPROVED** [review]
